andela/suyabay
# #116841693 Re-implementation of users endpoints

#### What does this PR do?

Re-implement user endpoints
#### Description of Task to be completed?

This PR Fix the following issues;
- Remove create user route and method from userController
- Fetch all users on Suyabay using this endpoint `api/v1/users`
- Create pagination on getting users endpoint  `api/v1/users?page=2`
- Create a new route to get a single user `api/v1/users/:username``
- Create a new route to get the current logged in user `api/v1/users/me`
- Create a new route for the current logged in user to update his data using put verb `api/v1/users/me`
- Create a new route for the current logged in user to update his data using patch verb `api/v1/users/me`
#### How should this be manually tested?
- To get all users, if you are using POSTMAN, change request method to Getting, type `api/v1/users` in the url  and click send you should see 

```
{
  "data": [
    {
      "id": 1,
      "username": "Sade adu",
      "email": "olotintemitop12e@gmail.com",
      "date_created": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-03-23 18:46:42.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": null
    },
    {
      "id": 2,
      "username": "Shania Hoeger",
      "email": "Rossie04@yahoo.com",
      "date_created": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": null
    },
    {
      "id": 3,
      "username": "Breanna Osinski II",
      "email": "Margie.West@yahoo.com",
      "date_created": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": null
    },
    {
      "id": 4,
      "username": "Terrence Wolff",
      "email": "Raymond13@Bernier.net",
      "date_created": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": null
    },
    {
      "id": 5,
      "username": "Edyth Stokes",
      "email": "Bins.Dell@gmail.com",
      "date_created": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-03-10 08:35:19.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": null
    },
    {
      "id": 6,
      "username": "Roy",
      "email": "temitope.olotin@andela.com",
      "date_created": {
        "date": "2016-03-10 08:41:34.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-03-23 19:17:28.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": "http://res.cloudinary.com/suyabay/image/upload/v1458735567/themzdqhwl8dpfxoutqx.jpg"
    },
    {
      "id": 9,
      "username": "laztopaz",
      "email": "olotintemitope@gmail.com",
      "date_created": {
        "date": "2016-03-23 18:15:39.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-04-03 16:02:27.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": null
    },
    {
      "id": 10,
      "username": "codekenn",
      "email": "kenny.ogunde@andela.com",
      "date_created": {
        "date": "2016-03-30 09:09:01.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-03-30 09:09:01.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": null
    }
  ]
}
```
- To get a user, if you are using POSTMAN, change request method to Getting, type `api/v1/users/Roy` in the url  and click send you should see

```
{
  "data": [
    {
      "id": 6,
      "username": "Roy",
      "email": "temitope.olotin@andela.com",
      "date_created": {
        "date": "2016-03-10 08:41:34.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "date_modified": {
        "date": "2016-03-23 19:17:28.000000",
        "timezone_type": 3,
        "timezone": "UTC"
      },
      "picture_url": "http://res.cloudinary.com/suyabay/image/upload/v1458735567/themzdqhwl8dpfxoutqx.jpg"
    }
  ]
}
```

otherwise see `{ "message" : "User Not found " }` 
- To get the next set of users using our pagination, simply set your request method to GET, then type 
  `api/v1/users?page=2` and click send if they are records you should see a JSON string listing the users otherwise you should see  `{ "data": [] }`
- To update the currently logged in user using PUT request, then type `api/v1/users/me` after that enter the details that you want to update into the form and press send. If it's successfully updated you should see `{ "message" : "User updated successfully " }` otherwise `{ "message" : "User Not found " }` 
- To update the currently logged in user using PATCH request, then type `api/v1/users/me` after that enter the details that you want to update into the form and press send. If it's successfully updated you should see `{ "message" : "User updated successfully " }` otherwise `{ "message" : "User Not found " }` 
#### Any background context you want to provide?
#### What are the relevant pivotal tracker stories?
#116841693
#### Screenshots (if appropriate)
#### Questions:

[@unicodeveloper](https://github.com/unicodeveloper)
